src/main/java/org/jenkinsci/plugins/saml/SamlProfileWrapper.java

```diff
@@ -48,13 +50,16 @@ public SamlProfileWrapper(SamlPluginConfig samlPluginConfig, StaplerRequest requ
     @SuppressWarnings("unused")
     @Override
     protected SAML2Profile process() {
-        SAML2Credentials credentials;
+        SAML2AuthenticationCredentials credentials;
         SAML2Profile saml2Profile;
         try {
             SAML2Client client = createSAML2Client();
             WebContext context = createWebContext();
-            credentials = (SAML2Credentials) client.getCredentials(context, JEESessionStore.INSTANCE).orElse(null);
-            saml2Profile = (SAML2Profile) client.getUserProfile(credentials, context, JEESessionStore.INSTANCE).orElse(null);
+            SessionStore sessionStore = createSessionStore();
+            CallContext ctx = new CallContext(context, sessionStore);
+            SAML2Credentials unvalidated = (SAML2Credentials) client.getCredentials(ctx).orElse(null);
+            credentials = (SAML2AuthenticationCredentials) client.validateCredentials(ctx, unvalidated).orElse(null);
```

Comment:
Per [the migration guide](https://www.pac4j.org/blog/what_s_new_in_pac4j_v6.html), the `getCredentials` method has been split into the `getCredentials` and `validateCredentials` methods.